lib/auth/facebook_authenticator.rb

```diff
@@ -10,158 +10,17 @@ def enabled?
     SiteSetting.enable_facebook_logins
   end
 
-  def description_for_user(user)
-    info = FacebookUserInfo.find_by(user_id: user.id)
-    info&.email || info&.username || ""
-  end
-
-  def can_revoke?
-    true
-  end
-
-  def revoke(user, skip_remote: false)
-    info = FacebookUserInfo.find_by(user_id: user.id)
-    raise Discourse::NotFound if info.nil?
-
-    if skip_remote
-      info.destroy!
-      return true
-    end
-
-    response = Excon.delete(revoke_url(info.facebook_user_id))
-
-    if response.status == 200
-      info.destroy!
-      return true
-    end
-
-    false
-  end
-
-  def revoke_url(fb_user_id)
-    "https://graph.facebook.com/#{fb_user_id}/permissions?access_token=#{SiteSetting.facebook_app_id}|#{SiteSetting.facebook_app_secret}"
-  end
-
-  def can_connect_existing_user?
-    true
-  end
-
-  def after_authenticate(auth_token, existing_account: nil)
-    result = Auth::Result.new
-
-    session_info = parse_auth_token(auth_token)
-    facebook_hash = session_info[:facebook]
-
-    result.email = email = session_info[:email]
-    result.email_valid = !email.blank?
-    result.name = facebook_hash[:name]
-
-    result.extra_data = facebook_hash
-
-    user_info = FacebookUserInfo.find_by(facebook_user_id: facebook_hash[:facebook_user_id])
-
-    if existing_account && (user_info.nil? || existing_account.id != user_info.user_id)
-      user_info.destroy! if user_info
-      result.user = existing_account
-      user_info = FacebookUserInfo.create!({ user_id: result.user.id }.merge(facebook_hash))
-    else
-      result.user = user_info&.user
-    end
-
-    if !result.user && !email.blank? && result.user = User.find_by_email(email)
-      FacebookUserInfo.create!({ user_id: result.user.id }.merge(facebook_hash))
-    end
-
-    user_info.update_columns(facebook_hash) if user_info
-
-    retrieve_avatar(result.user, result.extra_data)
-    retrieve_profile(result.user, result.extra_data)
-
-    if email.blank?
-      UserHistory.create(
-        action: UserHistory.actions[:facebook_no_email],
-        details: "name: #{facebook_hash[:name]}, facebook_user_id: #{facebook_hash[:facebook_user_id]}"
-      )
-    end
-
-    result
-  end
-
-  def after_create_account(user, auth)
-    extra_data = auth[:extra_data]
-    FacebookUserInfo.create!({ user_id: user.id }.merge(extra_data))
-
-    retrieve_avatar(user, extra_data)
-    retrieve_profile(user, extra_data)
-
-    true
-  end
-
   def register_middleware(omniauth)
     omniauth.provider :facebook,
            setup: lambda { |env|
              strategy = env["omniauth.strategy"]
               strategy.options[:client_id] = SiteSetting.facebook_app_id
               strategy.options[:client_secret] = SiteSetting.facebook_app_secret
-              strategy.options[:info_fields] = 'gender,email,name,about,first_name,link,last_name,website,location'
```

Comment:
all the gender,location etc is stuff we got to kill cause facebook do not provide it anymore?

Reply:
Yes, they have [tightened security a lot recently](https://developers.facebook.com/docs/facebook-login/changelog). Most of these fields now require formal App Review, and even then they require an additional `scope` parameter to be sent. So I am confident that no Discourse sites are currently using this functionality.